src/library_pthread.js

```diff
@@ -470,10 +470,10 @@ var LibraryPThread = {
       // Use Trusted Types compatible wrappers.
       if (typeof trustedTypes != 'undefined' && trustedTypes.createPolicy) {
         var p = trustedTypes.createPolicy('emscripten#workerPolicy2', { createScriptURL: (ignored) => pthreadMainJs });
-        worker = new Worker(p.createScriptURL('ignored'));
+        worker = new Worker(p.createScriptURL('ignored'){{{ EXPORT_ES6 ? ", {type: 'module'}" : '' }}});
```

Comment:
This syntax is pretty gross (i.e. hard to read), but I don't have a better idea

Reply:
Yeah I actually didn't even know it's supported like that and thought only about `#if EXPORT_ES6`, but that kind of multiline syntax is even worse :(

I really wish someday JS libraries would only use standard syntax and rely on optimisers evaluating settings at compile-time, but that seems like an unlikely pipe dream.